### DISCUSSION TITLE
Possible to create test module including Paparazzi?

### DISCUSSION THREAD

**Discussion by astrimagnussen** [cashapp/paparazzi#1056] [Q&A] [answered]
In the project Im working on, we have created an abstract class `PaparazziTest` containing all Paparazzi logic/setup needed for each test to run. Each test class then implement this class and write tests very simple, only providing what component should be tested with its setup. 
This base setup is now needed in multiple modules. My plan was to create a screenshot-testing module containing `PaparazziTest`. However, I cant seem to be able to use Paparazzi outside of the test folder? Is this by design, or can I adapt how imports are made to make Paparazzi available in src/main/java and not only src/test/java?

**Comment by TWiStErRob** [comment, marked as answer]
I have a module like that: https://github.com/TWiStErRob/net.twisterrob.sun/tree/4935be3f461e1cf3791ddf205bc5b8dd79183c6e/component/paparazzi

Used by other modules:
https://github.com/TWiStErRob/net.twisterrob.sun/blob/4935be3f461e1cf3791ddf205bc5b8dd79183c6e/component/widget/build.gradle.kts#L3

Via a custom plugin, but it doesn't do much:
https://github.com/TWiStErRob/net.twisterrob.sun/blob/4935be3f461e1cf3791ddf205bc5b8dd79183c6e/gradle/plugins/src/main/kotlin/net/twisterrob/sun/plugins/PaparazziPlugin.kt#L14-L17

---

I think the key is that we need to apply the paparazzi plugin to each module being tested, but in the shared module, there's no need to apply the plugin, just add the dependency that would be added as test-only by the paparazzi plugin:
https://github.com/cashapp/paparazzi/blob/09a33c0/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt#L303-L305
